crates/aggchain-proof-service/src/service.rs

```diff
@@ -137,38 +159,50 @@ impl tower::Service<AggchainProofServiceRequest> for AggchainProofService {
-                        l1_info_tree_merkle_proof: req.l1_info_tree_merkle_proof,
-                        l1_info_tree_leaf: req.l1_info_tree_leaf,
-                        l1_info_tree_root_hash: req.l1_info_tree_root_hash,
-                        ger_inclusion_proofs: req.ger_inclusion_proofs,
-                    };
-
-                proof_builder
-                    .call(aggchain_proof_builder_request)
-                    .map_err(Error::AggchainProofBuilderRequestFailed)
-                    .map(move |aggchain_proof_builder_result| {
-                        let agg_span_proof_response: AggchainProofBuilderResponse =
-                            aggchain_proof_builder_result?;
-                        Ok(AggchainProofServiceResponse {
-                            proof: agg_span_proof_response.proof,
-                            start_block: agg_span_proof_response.start_block,
-                            end_block: agg_span_proof_response.end_block,
-                            local_exit_root_hash: Default::default(),
-                            custom_chain_data: Default::default(),
-                        })
-                    })
+        async move {
+            // The ProposerResponse contains the start and end block number
+            // It also contains the generated proof.
+            let agg_span_proof_response = proposer_service
+                .call(proposer_request)
+                .await
+                .map_err(Error::ProposerServiceError)?;
+
+            let aggchain_proof_builder_request =
+                aggchain_proof_builder::AggchainProofBuilderRequest {
+                    agg_span_proof: agg_span_proof_response.agg_span_proof,
+                    start_block: agg_span_proof_response.start_block,
+                    end_block: agg_span_proof_response.end_block,
+                    l1_info_tree_merkle_proof: req.l1_info_tree_merkle_proof,
+                    l1_info_tree_leaf: req.l1_info_tree_leaf,
+                    l1_info_tree_root_hash: req.l1_info_tree_root_hash,
+                    ger_inclusion_proofs: req.ger_inclusion_proofs,
+                };
+
+            let aggchain_proof_response = proof_builder
+                .call(aggchain_proof_builder_request)
+                .await
+                .map_err(Error::AggchainProofBuilderServiceError)?;
+
+            let custom_chain_data = customchaindata_builder
+                .call(CustomChainDataBuilderRequest {
+                    l2_block_number: agg_span_proof_response.end_block,
+                    output_root: [1u8; 32],
+                })
+                .await
+                .map_err(Error::CustomChainDataBuilderError)?;
+
+            Ok(AggchainProofServiceResponse {
+                proof: aggchain_proof_response.proof,
+                start_block: agg_span_proof_response.start_block,
+                end_block: agg_span_proof_response.end_block,
+                local_exit_root_hash: Default::default(),
```

Comment:
Why `default`? Add it to the `AggchainProofBuilderResponse` and use it here.

Reply:
Because it was already `Default::default` and there is no `local_exit_root_hash` on the `AggchainProofBuilderResponse` If I'm not wrong